src/unix/linux/android/audio.c

```diff
@@ -60,18 +59,23 @@ static aaudio_data_callback_result_t audio_callback(AAudioStream *stream, void *
 	return AAUDIO_CALLBACK_RESULT_CONTINUE;
 }
 
-MTY_Audio *MTY_AudioCreate(uint32_t sampleRate, uint32_t minBuffer, uint32_t maxBuffer, uint8_t channels,
-	const char *deviceID, bool fallback)
+MTY_Audio *MTY_AudioCreate(MTY_AudioFormat format, uint32_t minBuffer,
+	uint32_t maxBuffer, const char *deviceID, bool fallback)
 {
 	MTY_Audio *ctx = MTY_Alloc(1, sizeof(MTY_Audio));
-	ctx->channels = channels;
-	ctx->sample_rate = sampleRate;
+	ctx->channels = format.channels;
+	ctx->channels_mask = format.channelsMask;
+	ctx->sample_format = format.sampleFormat;
+	ctx->sample_rate = format.sampleRate;
+	ctx->frame_size = format.channels *
+		(format.sampleFormat == MTY_AUDIO_SAMPLE_FORMAT_FLOAT ? sizeof(float) : sizeof(int16_t));
```

Comment:
you calculate this so many times by hand, that you might want to do it in a function, instead.